src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.tsx

```diff
@@ -151,9 +155,9 @@ function PopoverReportActionContextMenu(_props, ref) {
         isPinnedChat = false,
         isUnreadChat = false,
     ) => {
-        const nativeEvent = event.nativeEvent || {};
+        const {pageX = 0, pageY = 0} = 'nativeEvent' in event ? event.nativeEvent : {};
```

Comment:
How does `'nativeEvent' in event ? event.nativeEvent : ...` differ from `event?.nativeEvent ?? ...`

Reply:
TypeError: `Property 'nativeEvent' does not exist on type 'GestureResponderEvent | MouseEvent'.`. That's why we need `'nativeEvent' in event`

Reply:
Do you think that this would increase the readability?

```ts
// File-scoped util
function isGestureResponderEvent(event: GestureResponderEvent | MouseEvent): event is GestureResponderEvent {
    return 'nativeEvent' in event;
}

// ...

        // Usage:
        const {pageX = 0, pageY = 0} = isGestureResponderEvent(event) ? event.nativeEvent : {};

```

The type narrowing that is happening here wasn't clear to me until I investigated it in depth.

Reply:
Also, as we're fixing obvious bugs in this PR (at least judging from the current typing), it seems that the more defensive implementation for getting `{pageX, pageY}` would be...

```ts
function getPageCoords(event: GestureResponderEvent | MouseEvent): { readonly pageX: number, readonly pageY: number } {
    if ('nativeEvent' in event) {
        return event.nativeEvent;
    } else {
        return event;
    }
}

/// ...

// Usage:
        const {pageX, pageY} = getPageCoords(event);

```

Reply:
This is a great idea!